src/main/java/org/linksharing/server/http/ApiRestController.java

```diff
@@ -39,7 +45,16 @@ ResponseEntity<?> getProfileDetails(Principal user) {
     }
 
     @PostMapping("/profile")
-    ResponseEntity<?> updateProfileDetails(Principal user) {
+    ResponseEntity<?> updateProfileDetails(Principal user, @RequestBody UserProfileDetails profileDetails) {
+
+        UserProfileDetails userProfile = profileRepository.findByEmail(user.getName());
+
+        userProfile.setFirstName(profileDetails.getFirstName());
+        userProfile.setLastName(profileDetails.getLastName());
+        userProfile.setPublicEmail(profileDetails.getPublicEmail());
+
+        profileRepository.save(userProfile);
+
         return null;
```

Comment:
Should return user profile details data here.
The same is for `getProfileDetails`.
This data should be in html inputs, no additional divs or else is required.

Reply:
This one done. That was easier than I thought... Don't know how vue handles it, but looks like AI built in it)))